src/renderer/hooks/useInstantOnboarding.ts

```diff
@@ -0,0 +1,144 @@
+    )
+  }
+
+  const {
+    setShowInstantOnboarding,
+    setWelcomeQr,
+    showInstantOnboarding,
+    welcomeQr,
+  } = context
+
+  const switchToInstantOnboarding = useCallback(
+    async (qrWithUrl?: QrWithUrl) => {
+      if (qrWithUrl) {
+        if (
+          !['account', 'askVerifyGroup', 'askVerifyContact'].includes(
+            qrWithUrl.qr.kind
+          )
+        ) {
+          throw new Error(
+            'QR code needs to be of kind `account`, `askVerifyGroup` or `askVerifyContact` for instant onboarding flow'
+          )
+        }
+      }
+
+      setShowInstantOnboarding(true)
+
+      // Since a QR code can be scanned by the user in literally any situation,
+      // also when the user already has an existing account and is logged in, we
+      // can store it here temporarily, adjust the UI state and bring the user
+      // back to the QR code process again when everything is ready.
+      setWelcomeQr(qrWithUrl)
+    },
+    [setWelcomeQr, setShowInstantOnboarding]
+  )
+
+  const createInstantAccount = useCallback(
+    async (
+      accountId: number,
+      displayName: string,
+      profilePicture?: string
+    ): Promise<T.FullChat['id'] | null> => {
+      let instanceUrl = `dcaccount:${DEFAULT_CHATMAIL_INSTANCE_URL}`
+
+      // Use custom chatmail instance if given by QR code
+      if (welcomeQr && welcomeQr.qr.kind === 'account') {
+        instanceUrl = welcomeQr.url
+      }
+
+      // 1. In this "Instant Onboarding" account creation flow the user is not
+      // asked to manually insert any mail server credentials.
+      //
+      // Internally, the function will call the chatmail instance URL via HTTP
+      // from which it'll receive the address and password as an JSON object.
+      //
+      // After parsing it'll automatically configure the account with the
+      // appropriate keys for login, e.g. `addr` and `mail_pw`
+      await BackendRemote.rpc.setConfigFromQr(accountId, instanceUrl)
+
+      return new Promise((resolve, reject) => {
+        // 2. Kick-off the actual account creation process by calling
+        // `configure`. This happens inside of this dialog
+        openDialog(ConfigureProgressDialog, {
+          onSuccess: async () => {
+            try {
+              // 3. Additionally we set the `selfavatar` / profile picture
+              // and `displayname` configuration for this account.
+              //
+              // Note: We need to set these profile settings _after_
+              // calling `configure` to make the UI aware of them already
+              if (profilePicture) {
+                await BackendRemote.rpc.setConfig(
+                  accountId,
+                  'selfavatar',
+                  profilePicture
+                )
+              }
+              await BackendRemote.rpc.setConfig(
+                accountId,
+                'displayname',
+                displayName
+              )
```

Comment:
> Note: We need to set these profile settings _after_
> calling `configure` to make the UI aware of them already

what part of the UI does not support this? as both of these calls could fail I would prefer if we ran them before configuring (to not end up with a half configured account, or AEAP messages in device messages).

The problems I noticed when moving these two calls before calling configure (in this case opening the `ConfigureProgressDialog`):
- `window.__updateAccountListSidebar?.()` is missing in this onSuccess callback, without it the account list does not update reliably
- Core has a bug where the saved messages chat gets the own account avatar picture instead of the saved messages image, but that is a bug we should file and fix in core
- ... is there anything else that I missed in my testing?

Reply:
> what part of the UI does not support this?

That was my question as well. It just didn't work but I couldn't find out yet why, thanks for investigating.

Reply:
Filed the core issue here: https://github.com/deltachat/deltachat-core-rust/issues/5509